### ISSUE TITLE
PUSH: Received control message has a length limit

### ISSUE THREAD

**Issue by tomahkvt** [OpenVPN/openvpn#325]
**Describe the bug**
Received control message has a length limit of 2048 bytes. if we have more than 54 routes we reach this limit, the tail of this control message is cut. The tail of the control message contains information about ip for the tun interface and looks like 
`PUSH: Received control message: 'PUSH_REPLY,dhcp-option DNS  some_configuration_information  ifconfig 192.168.0.35 255.255.255'`

**To Reproduce**
Add to the OpenVPN server 60 routes. You will get the following error `Options error: ifconfig parms '192.168.0.35' and '255.255.255' must be valid addresses` trying to connect this server using OpenVPN client version 2.x.x. 

**Expected behavior**
It is expected that the control message length limit will be increased to 40960 bytes or, if possible, remove this limit.

**Version information (please complete the following information):**
 - OS: Ubuntu 20.04
 - OpenVPN client version: 2.4.7
 - All users who use OpenVPN client version 2.x.x have this problem.

**Comment by cron2** [comment]
There is no length limit in PUSH_REPLY messages - if there is "too many routes" they will normally be split across multiple PUSH_REPLY continuation messages.  I know that @ValdikSS has been sending many 1000s of routes to clients, and this works (though route installation became very slow then).

So this is either a problem in the process that generates the push routes (how are they generated on the server?  how does a more complete server log look like?  Note: push-options must be pushed one-by-one, not "all in one big push statement", for the continuation line handling to work), or a problem in the very old OpenVPN release that you are using.

2.4.7 is ancient software - 2.4.7 release was in February 2019, and since then, lots of bugs have been found and fixed.  Current OpenVPN release is 2.6.3, or if you want something from the previous stable branch, 2.5.9.

Note2: 2.4.x is technically end of support, so we are not going to spend much time to investigate bug reports in 2.4.x unless those can be reproduced in a current version.

**Comment by tomahkvt** [comment]
We just tested this case at OpenVPN versions 2.5.9 and 2.6.3. We have the same problem with Received control message length limit.

**Comment by cron2** [comment]
In that case, I assume there is a problem with the way you generate the to-be-pushed information.  As I wrote above, pushed information need to be generated one-by-one (`push "route 1"` - `push "route 2"` - `push "route 3"`, not `push "route 1 route 2 route 3"`).  Please provide information how you generate pushed config, and what the server log file looks like (verb 4, showing server startup and one client connecting)

**Comment by schwabe** [comment]
That is a known bug in Softether https://github.com/SoftEtherVPN/SoftEtherVPN/issues/1639. Is that the server software you are using?

**Comment by tomahkvt** [comment]
Guys, thank you a lot for your quick answer.
We are using AWS SAS https://docs.aws.amazon.com/vpn/latest/clientvpn-admin/what-is.html.
I didn't find information what version is in AWS.
Our control message looks like 
`PUSH: Received control message: 'PUSH_REPLY,dhcp-option DNS 172.16.0.1,dhcp-option DNS 172.16.0.2,route 192.168.1.2 255.255.255.0,.....`
Unfortunately, I can't provide how generates the control message
We tested openvpn3 and it works fine with this AWS VPN server. That is why we decided that the problem is with OpenVPN 2.x.x client.
